lib/spack/spack/test/cmd/install.py

```diff
@@ -502,6 +502,45 @@ def test_cdash_upload_extra_params(tmpdir, mock_fetch, install_mockery, capfd):
             assert '-my_custom_track' in content
 
 
+@pytest.mark.disable_clean_stage_check
```

Comment:
This seems to follow the established behavior for testing/handling reports and in that sense LGTM. Longer term (outside of this PR) I think it's worth considering why such an extensive test is required to make sure that the CDash report is properly constructed. Ideally, there would be a common interface among reporting classes, and you would just pass arguments in. For example, `cmd/install.py` already handles formulating a list of specs from a combination of user spec arguments and spec files.

Reply:
Yep, I definitely see your point there, thanks.